examples/video/src/main.rs

```diff
@@ -0,0 +1,265 @@
+            .api
+            .com
+            .atproto
+            .server
+            .get_service_auth(
+                atrium_api::com::atproto::server::get_service_auth::ParametersData {
+                    aud: VIDEO_SERVICE_DID.parse().expect("invalid DID"),
+                    exp: None,
+                    lxm: atrium_api::app::bsky::video::get_upload_limits::NSID.parse().ok(),
+                }
+                .into(),
+            )
+            .await?;
+        let client = AtpServiceClient::new(VideoClient::new(service_auth.data.token, None));
+        client.service.app.bsky.video.get_upload_limits().await?
+    };
+    println!("{:?}", limits.data);
+    if !limits.can_upload
+        || limits.remaining_daily_bytes.map_or(false, |remain| remain < data.len() as i64)
+        || limits.remaining_daily_videos.map_or(false, |remain| remain <= 0)
+    {
+        eprintln!("You cannot upload a video: {:?}", limits.data);
+        return Ok(());
+    }
+
+    // Upload video
+    println!("Uploading video...");
+    let output = {
+        let service_auth = agent
+            .api
+            .com
+            .atproto
+            .server
+            .get_service_auth(
+                atrium_api::com::atproto::server::get_service_auth::ParametersData {
+                    aud: format!(
+                        "did:web:{}",
+                        agent.get_endpoint().await.strip_prefix("https://").unwrap()
+                    )
+                    .parse()
+                    .expect("invalid DID"),
+                    exp: None,
+                    lxm: atrium_api::com::atproto::repo::upload_blob::NSID.parse().ok(),
+                }
+                .into(),
+            )
+            .await?;
+
+        let filename = args
+            .video
+            .file_name()
+            .and_then(|s| s.to_os_string().into_string().ok())
+            .expect("failed to get filename");
+        let client = AtpServiceClient::new(VideoClient::new(
+            service_auth.data.token,
+            Some(UploadParams { did: session.did.clone(), name: filename }),
+        ));
+        client.service.app.bsky.video.upload_video(data).await?
+    };
+    println!("{:?}", output.job_status.data);
+
+    // Wait for the video to be uploaded
+    let client = AtpServiceClient::new(ReqwestClient::new(VIDEO_SERVICE));
+    let mut status = output.data.job_status.data;
+    loop {
+        status = client
+            .service
+            .app
+            .bsky
+            .video
+            .get_job_status(
+                atrium_api::app::bsky::video::get_job_status::ParametersData {
+                    job_id: status.job_id.clone(),
+                }
+                .into(),
+            )
+            .await?
+            .data
+            .job_status
+            .data;
+        println!("{status:?}");
+        if status.blob.is_some()
+            || status.state == "JOB_STATE_CREATED"
```

Comment:
I think this is not correct, did you mean `JOB_STATE_COMPLETED`? https://docs.bsky.app/docs/api/app-bsky-video-upload-video

Reply:
Oops... thanks for pointing that out!